### PR TITLE
Operators support search tags.

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

Link to YouTrack:
[MAGN-7272](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7272) Add search tag support for built-in operators

Since operators are created in code, in dynamo core, I've added non-existent assemblies: `Operators` and `BuiltIn`. Dynamo will search for `Operators.xml` and `BuiltIn.xml` in root dynamo folder.

XML members will have names as `FunctionName`. I.e. "+" in XML will be "%add".

## NB
This PR will break operators and builtin icons!
image

I can fix it in this PR or create additional one.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate

### Reviewers

@holyjewsus am I right?

### FYIs

@lillismith

### CODE REVIEW

File: src/DynamoCore/Library/TypedParameter.cs [hunk 1]
```diff
@@ -42,7 +45,7 @@ public string Description
             get
             {
                 string description = string.Empty;
-                if (!string.IsNullOrEmpty(summary))
+                if (!string.IsNullOrEmpty(Summary))
```
Comment (aosyatnik) [inline src/DynamoCore/Library/TypedParameter.cs]: There is one regression now in master. Now we can't see parameter tooltips. 
I.e. that's what we see:
image

That's what we should see:
image

I fixed this regression here.

**Comment by aosyatnik** [comment]
@holyjewsus , please, don't merge it until tests will run without errors. I couldn't check it on Self CI.

File: src/DynamoCore/Library/DocumentationServices.cs [hunk 2]
```diff
@@ -46,13 +46,13 @@ public static XmlReader GetForAssembly(string assemblyPath, IPathManager pathMan
...
+            string cashedAssemblyLocation = assemblyLocation;
             if (pathManager != null)
                 pathManager.ResolveLibraryPath(ref assemblyLocation);
 
-            if (!File.Exists(assemblyLocation))
-            {
-                return false;
-            }
+            // Some nodes don't have assembly, e.g. operators, but they do have xml file.
+            if (String.IsNullOrEmpty(assemblyLocation))
```
Comment (holyjewsus) [inline src/DynamoCore/Library/DocumentationServices.cs]: just a style thing, but please use { } around the if statement here. (line 51)

File: src/DynamoCore/Library/XmlDocumentationExtensions.cs [hunk 3]
```diff
@@ -73,9 +73,6 @@ private static string CleanUpDocString(this string text)
         {
             var assemblyName = function.Assembly;
 
-            if (string.IsNullOrEmpty(assemblyName) || (function.Type == FunctionType.GenericFunction))
```
Comment (holyjewsus) [inline src/DynamoCore/Library/XmlDocumentationExtensions.cs]: are we sure that the assembly will never be null or empty?

**Comment by holyjewsus** [comment]
Thanks! @aosyatnik I've briefly tested this with an xml file named Operators.xml - and it worked nicely, is it possible to generate a template with all functionNames for @lillismith to fill in with search tags? (for both builtins and operators)

Have you considered how this will work with localization, is it as simple as adding another Operators.xml file to the en-us folder(for example)?

I also cannot run tests on the CI yet, we'll hold off merging until things calm down there.

**Comment by holyjewsus** [comment]
@aosyatnik also, sorry missed your comment above, why don't you do a fix for the icons in this PR.

**Comment by aosyatnik** [comment]
@holyjewsus , I didn't want to overload this PR, that's why I didn't add icons fix...

I've addressed your comments, added icons fix and added xml file for Operators.


About: "Have you considered how this will work with localization, is it as simple as adding another Operators.xml file to the en-us folder(for example)?"
Yes, if you need localization, you should add Operators.xml with translation in appropriate language folder.

**Comment by aosyatnik** [comment]
Hi, @holyjewsus , I've added xml file for BuiltIn nodes, that you sent me. Please, take a look one more time. I've added xml files and fixed icons.

**Comment by holyjewsus** [comment]
@aosyatnik I have yet to get a build to complete on the CI, but I'm going to merge this in, I've tested it in sandbox and it seems to work well.

**Pull request merged**